File: 575-distribute-candies/575-distribute-candies.py
class Solution:
    def distributeCandies(self, candyType: List[int]) -> int:
        candyTypes = set(candyType)
        edible = len(candyType) // 2
        edibleCandies = []
        count = 0
        for i in candyTypes:
            edibleCandies.append(i)
            count += 1
            if count == edible:
                break
        return len(edibleCandies)
        

File: 575-distribute-candies/575-distribute-candies.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        set<int> s;
        int edible, count;
        count=0;
        edible=candyType.size()/2;
        for(int item: candyType)
            s.insert(item);
        return min(edible, (int)s.size());
    }
};